src/Behat/Behat/Definition/Pattern/PatternTransformer.php

```diff
@@ -68,11 +71,16 @@ public function generatePattern($type, $stepText)
      */
     public function transformPatternToRegex($pattern)
     {
-        foreach ($this->policies as $policy) {
-            if ($policy->supportsPattern($pattern)) {
-                return $policy->transformPatternToRegex($pattern);
+        if (!isset($this->patternToRegexpCache[$pattern])) {
+            foreach ($this->policies as $policy) {
+                if ($policy->supportsPattern($pattern)) {
+                    $this->patternToRegexpCache[$pattern] = $policy->transformPatternToRegex($pattern);
+                    break;
+                }
             }
         }
+        return $this->patternToRegexpCache[$pattern];
```

Comment:
you should only return it if it's set (the loop maybe got to the end and didn't find anything)

Reply:
Note the unreachable exception after this - you could move the exception up and put a condition around it

Reply:
The fact the tests didn't catch this means there's one missing :-)

Reply:
🤦‍ Of course, it's fixed, and I added a test about it